lib/usagereporter/teleport/aggregating/reporter.go

```diff
@@ -321,3 +359,37 @@ func (r *Reporter) persistUserActivity(ctx context.Context, startTime time.Time,
 		}).Debug("Persisted user activity report.")
 	}
 }
+
+func (r *Reporter) persistResourceActivity(ctx context.Context, startTime time.Time, resourcePresences map[prehogv1.ResourceKind]map[string]struct{}) {
+	records := make([]*prehogv1.ResourceActivityRecord, 0, len(resourcePresences))
+	for kind, set := range resourcePresences {
+		record := &prehogv1.ResourceActivityRecord{}
+		record.ResourceKind = kind
+		record.ResourceName = make([][]byte, 0, len(set))
+		for name := range set {
+			record.ResourceName = append(record.ResourceName, r.anonymizer.AnonymizeNonEmpty(name))
```

Comment:
Check in with @bl-nero on what sort of truncation we want here. If we go for a `repeated fixed64` (i.e. a `[]uint64` on the go side) this might be as simple as the following.
```suggestion
			record.ResourceName = append(record.ResourceName, binary.LittleEndian.Uint64(r.anonymizer.AnonymizeNonEmpty(name)))
```